src/odil/DataSet.cpp

```diff
@@ -195,7 +195,7 @@ ::remove(Tag const & tag)
 {
     if(!this->has(tag))
     {
-        throw Exception("No such element");
+        throw Exception("No such element "+ tag.get_name());
```

Comment:
`Tag.get_name` may throw (e.g. for private tags): the safest way would be to use `operator std::string()` instead.